components/brave_perf_predictor/browser/p3a_bandwidth_savings_tracker.h

```diff
@@ -0,0 +1,38 @@
+/* Copyright 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_BRAVE_PERF_PREDICTOR_BROWSER_P3A_BANDWIDTH_SAVINGS_TRACKER_H_
+#define BRAVE_COMPONENTS_BRAVE_PERF_PREDICTOR_BROWSER_P3A_BANDWIDTH_SAVINGS_TRACKER_H_
+
+#include <array>
+#include <list>
+
+#include "base/timer/timer.h"
+
+class PrefRegistrySimple;
+class PrefService;
+
+namespace brave_perf_predictor {
+
+class P3ABandwidthSavingsTracker {
```

Comment:
do we really need this class? It just proxies a single call, right? Maybe we can merge it into the permanent state thing

Reply:
I've refactored to leave `P3ABandwidthSavingsPermanentState` only deal with permanent state, and `P3ABandwidthSavingsTracker` with reporting to a histogram too. The latter should also optimise how often data is loaded from permanent state and persisted to reduce profile writes

Reply:
@iefremov how does that sound?

Reply:
@AndriusA yep, looks good!